python/sidecar/typing.py

```diff
@@ -176,6 +178,28 @@ class ChatResponse(RefStudioModel):
     choices: list[ChatResponseChoice]
 
 
+class SearchRequest(RefStudioModel):
+    query: str
+    limit: int = 10
+
+
+class S2SearchResult(RefStudioModel):
+    title: str | None = None
+    abstract: str | None = None
+    venue: str | None = None
+    year: int | None = None
+    paperId: str | None = None
+    citationCount: int | None = None
+    openAccessPdf: str | None = None
+    authors: list[str] | None = None
+
+
+class SearchResponse(RefStudioModel):
+    status: ResponseStatus
+    message: str
+    results: list[S2SearchResult] | None = None
```

Comment:
Just noticed now that that this `| None` is making the field optional in typescript. That is bad developer experience. We should follow the convention other responses used of not having the `None` and making sure that we return an empty list on error (even though I don't expect anyone would use that).

Reply:
fixed 👍  - https://github.com/refstudio/refstudio/pull/313/commits/1305958fbcdaa55deacb4cf2358e0e8875d0bf33

Reply:
updated - https://github.com/refstudio/refstudio/pull/313/commits/3c96863c75602f900ea299b2f3f81aaab32eb6a8